File: src/test/java/com/github/kayjamlang/tests/containers/ContainerTest.java
package com.github.kayjamlang.tests.containers;

import com.github.kayjamlang.core.Stmt;
import com.github.kayjamlang.core.expressions.Expression;
import com.github.kayjamlang.core.KayJamLexer;
import com.github.kayjamlang.core.KayJamParser;
import com.github.kayjamlang.core.containers.Container;
import org.junit.BeforeClass;
import org.junit.Test;

import static org.junit.Assert.*;

public class ContainerTest {

    private static KayJamParser parser;

    @BeforeClass
    public static void prepare(){
        parser = new KayJamParser(new KayJamLexer("{true;false;var test = 12345;}"));
    }

    @Test
    public void test() throws Exception {
        Stmt stmt = parser.readStmt();

        assertNotNull(stmt);
        assertSame(Container.class, stmt.getClass());

        Container container = (Container) stmt;
        assertEquals(3, container.children.size());
    }
}

File: src/test/scala/com/github/kayjamlang/tests/containers/ContainerTest.scala
package com.github.kayjamlang.tests.containers

import com.github.kayjamlang.core.{KayJamLexer, KayJamParser}
import com.github.kayjamlang.core.containers.Container
import org.junit.Assert._
import org.junit.{BeforeClass, Test}


object ContainerTest {
  private var parser: KayJamParser = null

  @BeforeClass def prepare(): Unit = {
    parser = new KayJamParser(new KayJamLexer(
      """
        |{
        | true;
        | false;
        | var test = 12345;
        |}
        |""".stripMargin))
  }
}

class ContainerTest {
  @Test
  @throws[Exception]
  def test(): Unit = {
    val expression = ContainerTest.parser.readExpression
    assertNotNull(expression)
    assertSame(classOf[Container], expression getClass)
    val container = expression.asInstanceOf[Container]
    assertEquals(3, container.children size)
  }
}